src/supervisor/watchers/handlers/pod.ts

```diff
@@ -38,6 +38,10 @@ export async function handleReadyPod(
     // ImageName may contain a tag. The image behind this tag can be mutated and can change over time.
     // We need to compare on ImageID which will reliably tell us if the image has changed.
     if (scanned === workload.imageId) {
+      logger.debug(
+        { scanned: scanned, imageId: workload.imageId },
+        'image already scanned',
+      );
       continue;
     }
     await setWorkloadImageAlreadyScanned(
```

Comment:
Should we push the `workload` to `workloadToScan` first before `setWorkloadImageAlreadyScanned`? Or does it not make much time difference?

Reply:
Time difference in what sense? The `await` is just future-proofing the function in case we store this data in an external cache 😄 Since it now sets data in memory it's pretty much as fast as it gets

Reply:
ok. well, I don't know what the time difference would be. But in the case of the order, I thought it should be pushed to the  `workloadToScan`, and then marked the workload as `setWorkloadImageAlreadyScanned`. I guess the scanned status and the `workloadToScan` is totally two different thing, so the order here doesn't really matter :)

Reply:
In this case the order probably doesn't matter because the actions happen one after another and should produce identical results.

If these were operations that could yield the event loop then the order would matter. If we pushed to the queue and tried to store the image to the cache, but that yielded the loop and caused a similar task for the same workload to be run again, then we could risk pushing the same workload to the queue twice. https://snyk.io/blog/nodejs-how-even-quick-async-functions-can-block-the-event-loop-starve-io/ (the "👋 Micro-Tasks" section) Since our `await` doesn't do that here, the order won't matter 😄